### ISSUE TITLE
stb_c_lexer crash

### ISSUE THREAD

**Issue by louisliangjun** [nothings/stb#1118]
crash when parse source NOT endswith newline.

// main helper function for returning a parsed token
static int stb__clex_token(stb_lexer *lexer, int token, char *start, char *end)
{
   lexer->token = token;
   lexer->where_firstchar = start;
   lexer->where_lastchar = end;
   lexer->parse_point = end+1;    <---- here,  fix ---->      (end == lexer->eof) ? end : end+1
   return 1;
}

**Comment by rygorous** [comment]
Ugh, it looks like stb_c_lexer has never actually been tested on non-NUL-terminated input strings, and has buffer overruns in most paths when handed such an input.

Until this is resolved, I would recommend only using stb_c_lexer on inputs that have a terminating NUL added.

**Comment by louisliangjun** [comment]
My test already used NUL-terminated string... it maybe another bug         
            
                
                    
                            
                                
                            
                            
                                jun louis
                            
                    
                        
                            
                                    ***@***.***
                            
                        
                
            
        ---- Replied Message ***@***.***>Date07/03/2021 08:16 ***@***.***> ***@***.******@***.***>SubjectRe: [nothings/stb] stb_c_lexer crash (#1118)
Ugh, it looks like stb_c_lexer has never actually been tested on non-NUL-terminated input strings, and has buffer overruns in most paths when handed such an input.
Until this is resolved, I would recommend only using stb_c_lexer on inputs that have a terminating NUL added.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/nothings/stb/issues/1118#issuecomment-873312849",
"url": "https://github.com/nothings/stb/issues/1118#issuecomment-873312849",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by nothings** [comment]
Yes, you're right, I think it's just the bug you describe in stb__clex_token.

**Comment by johnmanthop** [comment]
I would like to work on that, is there already a pull request on that? If not, is there anything else I should keep in mind?

**Comment by louisliangjun** [comment]
I simple fix it use

lexer->parse_point = end+1; <---- here, fix ----> (end == lexer->eof) ? end : end+1


> I would like to work on that, is there already a pull request on that? If not, is there anything else I should keep in mind?
